ganttproject/src/main/java/net/sourceforge/ganttproject/ProjectStateHolderEventListener.kt

```diff
@@ -19,24 +19,72 @@ along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
 
 package net.sourceforge.ganttproject
 
-import biz.ganttproject.app.Barrier
-import biz.ganttproject.app.BarrierEntrance
 import net.sourceforge.ganttproject.ProjectEventListener.Stub
 import net.sourceforge.ganttproject.storage.ProjectDatabase
+import net.sourceforge.ganttproject.task.event.*
 
 /**
  * Holds the current state of a Gantt project, updating it on events.
  *
- * @param databaseFactory - factory for generating a project state storage.
+ * @param projectDatabase - database which holds the current project state.
  */
-class ProjectStateHolderEventListener(private val databaseFactory: () -> ProjectDatabase) : Stub() {
-  private val lazyProjectDatabase = lazy { databaseFactory()  }
-  private val projectDatabase: ProjectDatabase by lazyProjectDatabase
-
-  override fun projectClosed() {
-    if (lazyProjectDatabase.isInitialized()) {
-      // ...
-      projectDatabase.shutdown()
+class ProjectStateHolderEventListener(private val projectDatabase: ProjectDatabase) : TaskListener, Stub() {
```

Comment:
Seems like there're a lot of listeners yet to add, so it's better to create a separate class for each interface, passing the same db, right?

Reply:
Should we modify it later if required?